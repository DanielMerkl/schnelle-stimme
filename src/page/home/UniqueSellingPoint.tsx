import React, { FC } from 'react';
import styled from 'styled-components';
import { Typography } from '@material-ui/core';

interface Props {
  svg: string;
  alt: string;
  gridArea: 'first' | 'second' | 'third';
  headline: string;
  subtext: string;
}

export const UniqueSellingPoint: FC<Props> = ({
  svg,
  alt,
  gridArea,
  headline,
  subtext,
}) => (
  <Wrapper gridArea={gridArea}>
    <StyledImg src={svg} alt={alt} />
    <Typography variant="h5" align="center">
      {headline}
    </Typography>
    <Typography align="center">{subtext}</Typography>
  </Wrapper>
);

interface WrapperProps {
  gridArea: string;
}

const Wrapper = styled.section<WrapperProps>`
  max-width: 400px;
  grid-area: ${(props) => props.gridArea};
  display: grid;
  gap: 0.5rem;
  grid-template-rows: auto auto 1fr;
  justify-items: center;
`;

const StyledImg = styled.img`
  max-width: 150px;
`;
