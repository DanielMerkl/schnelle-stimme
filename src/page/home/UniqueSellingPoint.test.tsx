import React from 'react';
import { render } from '@testing-library/react';

import { UniqueSellingPoint } from './UniqueSellingPoint';

describe('UniqueSellingPoint', () => {
  it('renders correctly', () => {
    const { container } = render(
      <UniqueSellingPoint
        svg="picture.svg"
        alt="Description of the picture."
        gridArea="first"
        headline="Banana"
        subtext="Bananas are not always crooked."
      />
    );

    expect(container).toMatchSnapshot();
  });
});
