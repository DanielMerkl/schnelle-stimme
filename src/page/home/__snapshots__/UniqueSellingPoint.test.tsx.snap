// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniqueSellingPoint renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root sc-AxjAm eimCaO MuiPaper-elevation1 MuiPaper-rounded"
    gridarea="first"
  >
    <img
      alt="Description of the picture."
      class="sc-AxirZ hyvKOs"
      src="picture.svg"
    />
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
    >
      Banana
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      Bananas are not always crooked.
    </p>
  </div>
</div>
`;
